test/rpc_test.py

```diff
@@ -533,6 +534,59 @@ def test_py_user_defined(self):
         )
         self.assertEqual(ret, my_function(n, n + 1, n + 2))
 
+    def _profiler_test_with_rpc(self, rpc_exec_mode, func, args):
+        dst = (self.rank + 1) % self.world_size
+        # only run profiler on rank 0.
+        if self.rank in {0}:
+            with torch.autograd.profiler.profile() as prof:
+                if rpc_exec_mode == RPCExecMode.SYNC:
+                    rpc.rpc_sync("worker{}".format(dst), func, args=args)
+                elif rpc_exec_mode == RPCExecMode.ASYNC:
+                    fut = rpc.rpc_async("worker{}".format(dst), func, args=args)
+                    fut.wait()
+                else:
+                    self.assertTrue(rpc_exec_mode == RPCExecMode.REMOTE)
+                    rref = rpc.remote("worker{}".format(dst), func, args=args)
+                    rref.to_here()
+                    # We need to wait for the instance to be created on
+                    # the owner, and get back a positive confirmation.
+                    # Calling to_here does not ensure that we have finished
+                    # processing the Owner's confirmation of this RRef. To do
+                    # this, we wait until the current RRef context doesn't have
+                    # any pending users, which indicates that the confirmation
+                    # was processed on this worker.
+                    num_pending_users = int(_rref_context_get_debug_info()["num_pending_users"])
```

Comment:
After speaking to @mrshenli, there's no way to currently verify that we've received a confirmation message from the node that owns the RRef in the protocol. We may need to add this in the future. 

I don't think the user should do this when using the profiler in their application, but the risk we currently run is if we immediately exit the profiler after creating an RRef, it may not show up in the profiler result. Things that will help with this is adding a public API and making `to_here()` not block a thread: https://github.com/pytorch/pytorch/issues/31461

Reply:
I think this would work for testing, although is not an ideal public API. We could add an `RRef.wait()` API if that's generally helpful.